artichoke-backend/src/extn/core/string/mruby.rs

```diff
@@ -213,7 +213,17 @@ unsafe extern "C" fn string_byteslice(mrb: *mut sys::mrb_state, slf: sys::mrb_va
     let length = length.map(Value::from);
     let result = trampoline::byteslice(&mut guard, value, index, length);
     match result {
-        Ok(value) => value.inner(),
+        Ok(value) if value.is_nil() => value.inner(),
```

Comment:
`nil` values are structured differently and if we don't short circuit here, attempting to set the class pointer on a `nil` will segfault.